src/token/fedauth-info-parser.js

```diff
@@ -0,0 +1,62 @@
+const async = require('async');
+
+var readFedAuthInfoOpt = (data, infoData, callback) => {
+  const FEDAUTHINFOID = {
+    STSURL: 0x01,
+    SPN: 0x02
+  };
+  const fedauthInfoID = data.readUInt8(infoData.offset);
+  infoData.offset += 1;
+  const fedAuthInfoDataLen = data.readUInt32LE(infoData.offset);
+  infoData.offset += 4;
+  const fedAuthInfoDataOffset = data.readUInt32LE(infoData.offset);
+  infoData.offset += 4;
+  switch (fedauthInfoID) {
+    case FEDAUTHINFOID.SPN:
+      infoData.spnLen = fedAuthInfoDataLen;
+      infoData.spnOffset = fedAuthInfoDataOffset;
+      break;
+    case FEDAUTHINFOID.STSURL:
+      infoData.stsurlLen = fedAuthInfoDataLen;
+      infoData.stsurlOffset = fedAuthInfoDataOffset;
+      break;
+              // ignoring unknown fedauthinfo options
+    default:
+      break;
+  }
+
+  if (infoData.spnLen && infoData.stsurlLen) {
+    const spn = data.slice(infoData.spnOffset, infoData.spnOffset + infoData.spnLen);
```

Comment:
Instead of slicing out a new `Buffer`, and then converting the `Buffer` to a string via `Buffer.prototype.toString`, I think we could use `Buffer.prototype.toString` directly. Also, I think by specifying the correct encoding (`ucs2`), we could get rid of the `String.prototype.replace` call to filter out the null bytes:

```js
const spn = data.toString('ucs2', infoData.spnOffset, infoData.spnOffset + infoData.spnLen);
const stsurl = data.toString('ucs2', infoData.stsurlOffset, infoData.stsurlOffset + infoData.stsurlLen);
```